src/main/java/com/mongodb/spark/sql/connector/read/MongoScanBuilder.java

```diff
@@ -272,23 +281,30 @@ private FilterAndPipelineStage processFilter(final Filter filter) {
       }
     } else if (filter instanceof StringContains) {
       StringContains stringContains = (StringContains) filter;
+      String fieldName = getFieldName(stringContains.attribute());
       return new FilterAndPipelineStage(
-          filter,
-          Filters.regex(stringContains.attribute(), format(".*%s.*", stringContains.value())));
+          filter, Filters.regex(fieldName, format(".*%s.*", stringContains.value())));
     } else if (filter instanceof StringEndsWith) {
       StringEndsWith stringEndsWith = (StringEndsWith) filter;
+      String fieldName = getFieldName(stringEndsWith.attribute());
       return new FilterAndPipelineStage(
-          filter,
-          Filters.regex(stringEndsWith.attribute(), format(".*%s$", stringEndsWith.value())));
+          filter, Filters.regex(fieldName, format(".*%s$", stringEndsWith.value())));
     } else if (filter instanceof StringStartsWith) {
       StringStartsWith stringStartsWith = (StringStartsWith) filter;
+      String fieldName = getFieldName(stringStartsWith.attribute());
       return new FilterAndPipelineStage(
-          filter,
-          Filters.regex(stringStartsWith.attribute(), format("^%s.*", stringStartsWith.value())));
+          filter, Filters.regex(fieldName, format("^%s.*", stringStartsWith.value())));
     }
     return new FilterAndPipelineStage(filter, null);
   }
 
+  private String getFieldName(final String attribute) {
+    if (attribute.startsWith("`") && attribute.endsWith("`")) {
```

Comment:
```suggestion
    // Spark automatically escapes hyphenated names using backticks
    if (attribute.startsWith("`") && attribute.endsWith("`")) {
```
Could also rename: 
`toJavaFieldName(final String sparkAttribute)`